components/match2/wikis/trackmania/match_group_input_custom.lua

```diff
@@ -0,0 +1,480 @@
+		-- parse player
+		local player = Json.parseIfString(match['opponent' .. opponentIndex .. '_p' .. playerIndex]) or {}
+		player.name = player.name or Variables.varDefault(teamName .. '_p' .. playerIndex)
+		player.displayname = player.displayname
+			or Variables.varDefault(teamName .. '_p' .. playerIndex .. 'dn', player.name:gsub('_', ' '))
+		player.flag = player.flag or Variables.varDefault(teamName .. '_p' .. playerIndex .. 'flag')
+		if not Table.isEmpty(player) then
+			match['opponent' .. opponentIndex .. '_p' .. playerIndex] = player
+		end
+	end
+	return match
+end
+
+--
+-- map related functions
+--
+function mapFunctions.getExtraData(map)
+	local overtimes = Array.extractValues(Table.mapValues(mw.text.split(map.overtime or '', ','), tonumber))
+
+	map.extradata = {
+		comment = map.comment,
+		header = map.header,
+		overtime = Table.isNotEmpty(overtimes) and Json.stringify(overtimes) or nil
+	}
+	return map
+end
+
+function mapFunctions.getScoresAndWinner(map)
+	map.scores = {}
+	local indexedScores = {}
+	for scoreIndex = 1, MAX_NUM_OPPONENTS do
+		-- read scores
+		local score = map['score' .. scoreIndex]
+		local obj = {}
+		if not Logic.isEmpty(score) then
+			if TypeUtil.isNumeric(score) then
+				score = tonumber(score)
+				obj.status = STATUS_HAS_SCORE
+				obj.score = score
+				obj.index = scoreIndex
+			elseif Table.includes(ALLOWED_STATUSES, score) then
+				obj.status = score
+				obj.score = -1
+				obj.index = scoreIndex
+			end
+			table.insert(map.scores, score)
+			indexedScores[scoreIndex] = obj
+		else
+			break
+		end
+	end
+	local isFinished = map.finished
+	if not Logic.isEmpty(isFinished) then
+		isFinished = Logic.readBool(isFinished)
+	else
+		isFinished = not Logic.readBool(map.unfinished)
+	end
+	if isFinished and not Logic.isEmpty(indexedScores) then
+		map.winner = mapFunctions.getWinner(indexedScores)
+	end
+
+	return map
+end
+
+function mapFunctions.getWinner(indexedScores)
+	table.sort(indexedScores, mapFunctions.mapWinnerSortFunction)
+	return indexedScores[1].index
+end
+
+function mapFunctions.mapWinnerSortFunction(op1, op2)
+	local op1norm = op1.status == STATUS_HAS_SCORE
+	local op2norm = op2.status == STATUS_HAS_SCORE
+	if op1norm then
+		if op2norm then
+			return tonumber(op1.score) > tonumber(op2.score)
+		else return true end
+	else return CustomMatchGroupInput._sortOpponents(op1, op2, op1norm, op2norm) end
+end
+
+function mapFunctions.getTournamentVars(map)
+	map.mode = Logic.emptyOr(map.mode, Variables.varDefault('tournament_mode', '2v2'))
+	return MatchGroupInput.getCommonTournamentVars(map)
+end
+
+function mapFunctions.getParticipantsData(map)
+	local participants = map.participants or {}
+
+	for opponentIndex = 1, MAX_NUM_OPPONENTS do
+		for player = 1, MAX_NUM_PLAYERS do
+			local participant = participants[opponentIndex .. '_' .. player] or {}
+			if not Table.isEmpty(participant) then
+				participants[opponentIndex .. '_' .. player] = participant
+			end
```

Comment:
This does nothing?
```
a = b
if a then
  b = a
end
```

Reply:
Should we just remove `mapFunctions.getParticipantsData` and its usages ?

Reply:
If I understand well there is no reason to keep this function as this snippet just does nothing

Reply:
https://github.com/Liquipedia/Lua-Modules/pull/2273/commits/8c39bcc2591a52f97087d59d0a642f01bf287fde